lib/elixir/lib/io.ex

```diff
@@ -233,20 +233,39 @@ defmodule IO do
   @doc """
   Inspects and writes the given `item` to the device.
 
+  It's important to note that it returns the given `item` unchanged.
+  This makes it possible to insert an `IO.inspect/2` call into
+  virtually any place of the code or in the middle of a pipeline
+  to "spy" on the performed transformations.
+
   It enables pretty printing by default with width of
   80 characters. The width can be changed by explicitly
   passing the `:width` option.
 
   The output can be decorated with a tag, by providing a `:tag`
   option to easily distinguish it from other `IO.inspect/2` calls.
+  The tag will be printed before the inspected `item`.
 
   See `Inspect.Opts` for a full list of remaining formatting options.
 
   ## Examples
 
-      IO.inspect Process.list, width: 40
-
-      IO.inspect Node.list, tag: "nodes"
+      IO.inspect <<0, 1, 2>>, width: 40
+      #=> <<0, 1, 2>>
+      <<0, 1, 2>>
+
+      IO.inspect 1..100, tag: "a wonderful range"
+      #=> a wonderful range: 1..100
+      1..100
+
+      [1, 2, 3]
+      |> IO.inspect(tag: "before")
+      |> Enum.map(&(&1 * 2))
+      |> IO.inspect(tag: "after")
+      |> Enum.sum
+      #=> before: [1, 2, 3]
+      #=> after: [2, 4, 6]
+      12
```

Comment:
Awesome with the before/after example, I love it. About the format though, what if we do:

```elixir
@doc """
...

## Examples

    IO.inspect <<0, 1, 2>>, width: 40
    # Prints:
    # <<0, 1, 2>>

    IO.inspect 1..100, tag: "a wonderful range"
    # Prints:
    # a wonderful range: 1..100
```

without showing the return value (which is what we already do?